cmd/restic/exclude.go

```diff
@@ -385,6 +387,33 @@ func rejectBySize(maxSizeStr string) (RejectFunc, error) {
 	}, nil
 }
 
+// rejectCloudFiles returns a func which rejects files which are online-only cloud files
+func rejectCloudFiles() (RejectFunc, error) {
+	if runtime.GOOS != "windows" {
+		return nil, errors.Fatalf("exclude-cloud-files is only supported on Windows")
+	}
+
+	const (
+		FILE_ATTRIBUTE_UNPINNED              = 0x00100000
+		FILE_ATTRIBUTE_RECALL_ON_DATA_ACCESS = 0x00400000
+	)
+
+	return func(item string, fi os.FileInfo) bool {
+		attrs, ok := fi.Sys().(*syscall.Win32FileAttributeData)
```

Comment:
Please move the fs specific code into the `internal/fs` package (maybe `internal/fs/file_windows.go`) and add a minimal unit test there. I'd like to minimize new code in `cmd/restic` as much as reasonable.

Reply:
The unit test will probably have to use a bit of mocking to create an os.FileInfo that pretends to be a cloud file, but shouldn't be too complex to implement.

Reply:
I've given this a go, but I'm not 100% sure it's right, I've never done multi-targeting with go before.

I believe I had to create a stub implementation in `internal/fs/file_unix.go`, then the real implementation in `internal/fs/file_windows.go`.

I've added 3 tests
1. Test with a real temp file that the function doesn't err
2. Mock cloud file
3. Mock regular file